pkg/frontend/clustermanager_putorpatch_test.go

```diff
@@ -226,3 +227,73 @@ func TestPutOrPatchClusterManagerConfiguration(t *testing.T) {
 		})
 	}
 }
+
+var populatedSystemData = api.SystemData{
+	LastModifiedAt:     &time.Time{},
+	LastModifiedBy:     "test-user",
+	LastModifiedByType: api.CreatedByTypeUser,
+	CreatedAt:          &time.Time{},
+	CreatedBy:          "test-user",
+	CreatedByType:      api.CreatedByTypeUser,
+}
+
+func TestEnrichSyncSetSystemData(t *testing.T) {
+	doc := api.ClusterManagerConfigurationDocument{
+		SyncSet: &api.SyncSet{},
+	}
+
+	enrichSyncSetSystemData(&doc, &populatedSystemData)
+
+	compareSystemData(t, *doc.SyncSet.SystemData, populatedSystemData)
+}
+
+func TestEnrichSyncIdentityProviderSystemData(t *testing.T) {
+	doc := api.ClusterManagerConfigurationDocument{
+		SyncIdentityProvider: &api.SyncIdentityProvider{},
+	}
+
+	enrichSyncIdentityProviderSystemData(&doc, &populatedSystemData)
+
+	compareSystemData(t, *doc.SyncIdentityProvider.SystemData, populatedSystemData)
+}
+
+func TestEnrichMachinePoolSystemData(t *testing.T) {
+	doc := api.ClusterManagerConfigurationDocument{
+		MachinePool: &api.MachinePool{},
+	}
+
+	enrichMachinePoolSystemData(&doc, &populatedSystemData)
+
+	compareSystemData(t, *doc.MachinePool.SystemData, populatedSystemData)
+}
+
+func TestEnrichSecretSystemData(t *testing.T) {
+	doc := api.ClusterManagerConfigurationDocument{
+		Secret: &api.Secret{},
+	}
+
+	enrichSecretSystemData(&doc, &populatedSystemData)
+
+	compareSystemData(t, *doc.Secret.SystemData, populatedSystemData)
+}
+
+func compareSystemData(t *testing.T, docSystemData, expectedSystemData api.SystemData) {
+	if docSystemData.CreatedAt == nil || docSystemData.CreatedAt != expectedSystemData.CreatedAt {
+		t.Fatalf("CreatedAt was %q expected %q", docSystemData.CreatedAt, expectedSystemData.CreatedAt)
```

Comment:
Suggestion for readability, but totally optional - might make more sense to fail the test inside of the `Test` functions themselves rather than in this external one. We would still probably want to return these specific errors rather than one single nonspecific error, and then fail the tests if `err != nil`.

Reply:
I'll make this change, thanks for the suggestion